mesonbuild/interpreter/type_checking.py

```diff
@@ -530,9 +530,24 @@ def link_whole_validator(values: T.List[T.Union[StaticLibrary, CustomTarget, Cus
     NATIVE_KW,
 ]
 
+
+def _name_validator(arg: T.Optional[T.Union[str, T.List]]) -> T.Optional[str]:
+    if isinstance(arg, list) and arg:
+        return 'may be an empty array to signify default.'
+    return None
+
+
+_NAME_PREFIX_KW: KwargInfo[T.Optional[str]] = KwargInfo(
```

Comment:
Is the type argument correct with the ContainerTypeInfo below?

Reply:
Good question... it returns Optional[str], but I guess it takes `list | str`....